### PR TITLE
feat(experimental): `InputPhoneInternational` refactor to new textfield approach

### PR DESCRIPTION
**Pull request by waterplea**
Fixes #

### CODE REVIEW

**Review by splincode — APPROVED** [review]

File: projects/experimental/components/input-phone-international/input-phone-international.style.less [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+
+.t-ipi-code {
+    color: var(--tui-text-secondary);
+}
+
+.t-ipi-chevron {
+    font-size: 1rem;
+}
+
+.t-ipi-search {
```
Comment (nsbarsukov) [inline projects/experimental/components/input-phone-international/input-phone-international.style.less:97]: https://github.com/user-attachments/assets/e05fde32-504a-4381-a39e-b9eb79aad0a0

File: projects/demo/src/modules/components/input-phone-international/index.html [hunk 2]
```diff
@@ -226,10 +221,18 @@
...
-                    Type a phone number
-                </tui-input-phone-international>
+                    <label tuiLabel>Type a phone number</label>
+                    <input
+                        tuiInputPhoneInternational
+                        [countries]="countries"
+                        [countrySearch]="countrySearch"
+                        [formControl]="formControl"
+                        [invalid]="control.invalid"
+                        [readOnly]="control.readonly"
```
Comment (nsbarsukov) [inline projects/demo/src/modules/components/input-phone-international/index.html:232]: https://github.com/user-attachments/assets/523952e6-e1bb-46d4-bd95-4776f9d70844

File: projects/demo/src/modules/components/input-phone-international/index.html [hunk 3]
```diff
@@ -226,10 +221,18 @@
                     [tuiDropdownOffset]="dropdown.offset"
                     [tuiTextfieldCleaner]="textfield.cleaner"
                     [tuiTextfieldSize]="textfield.size"
-                    [(countryIsoCode)]="countryIsoCode"
                 >
-                    Type a phone number
-                </tui-input-phone-international>
+                    <label tuiLabel>Type a phone number</label>
+                    <input
```
Comment (nsbarsukov) [inline projects/demo/src/modules/components/input-phone-international/index.html:226]: Add `disabled` too

https://taiga-previews-demo--pr9976-input-phone-international-6lw0j9ex.web.app/components/input-phone-international/API?disabled=true

File: projects/experimental/components/input-phone-international/input-phone-international.template.html [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+        ngModel=""
+        tabindex="-1"
+        tuiTextfield
+        class="t-ipi-select"
+        [attr.data-focus]="open()"
+        [attr.data-mode]="mode()"
+        [disabled]="disabled()"
+        [focused]="false"
+        [ngModelOptions]="{standalone: true}"
+        (mousedown.prevent)="interactive() && open.set(!open())"
```
[the remark this answers is not in the document]
Comment (waterplea) [reply projects/experimental/components/input-phone-international/input-phone-international.template.html:12]: Interesting. Currently input is always focused in this new version. I wonder if we can dynamically control keyboard from none to digits depending on the openness of the dropdown. We also need to try the mobile dropdown with this, I'll make an example and try all that.

File: projects/demo/src/modules/components/input-phone-international/index.html (hunk 2 above)
Comment (waterplea) [reply projects/demo/src/modules/components/input-phone-international/index.html:232]: Looks like we do not have readonly design for dark mode in new textfields also. Writing it here just so I don't forget to fix it

File: projects/experimental/components/input-phone-international/input-phone-international.template.html [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+        </tui-textfield>
+    </div>
+
+    <tui-data-list class="t-ipi-options">
+        <button
+            *ngFor="let item of filtered()"
+            tuiCell
+            tuiOption
+            type="button"
+            (click)="onItemClick(item.iso)"
```
Comment (nsbarsukov) [inline projects/experimental/components/input-phone-international/input-phone-international.template.html:55]: https://github.com/user-attachments/assets/d183f257-5f92-4b78-aab9-2938b4f0d72b

File: projects/demo/src/modules/components/input-phone-international/index.html [hunk 6]
```diff
@@ -206,17 +205,43 @@
...
+                <strong>Maskito</strong>
+                to format phone number.
+                <br />
+                Don't hesitate to use it too to manually format any phone number.
+            </ng-template>
+        </tui-doc-example>
+
+        <tui-doc-example
+            id="mobile-dropdown"
+            heading="Mobile dropdown"
```
Comment (nsbarsukov) [inline projects/demo/src/modules/components/input-phone-international/index.html:228]: https://github.com/user-attachments/assets/8e7a9329-4572-407e-afce-1e0b89027287

Iphone 13 mini, IOS 18

File: projects/demo-playwright/tests/legacy/input-phone-international/input-phone-international.pw.spec.ts [hunk 7]
```diff
@@ -11,7 +11,7 @@ test.describe('InputPhoneInternational', () => {
 
         test.beforeEach(({page}) => {
             example = new TuiDocumentationPagePO(page).apiPageExample;
-            select = example.locator('tui-input-phone-international select');
+            select = example.locator('tui-textfield .t-ipi-select');
```
Comment (nsbarsukov) [inline projects/demo-playwright/tests/legacy/input-phone-international/input-phone-international.pw.spec.ts:14]: Let's create `input-phone-international.po.ts` inside [projects/demo-playwright/utils/page-objects](https://github.com/taiga-family/taiga-ui/tree/main/projects/demo-playwright/utils/page-objects):

```ts
import type {Locator} from '@playwright/test';

import {TuiTextfieldWithDataListPO} from './textfield-with-data-list.po';

export class TuiInputPhoneInternationalPO extends TuiTextfieldWithDataListPO {
    public select: Locator = this.host.locator('tui-textfield .t-ipi-select');
}
```

And use it:

```ts
let inputPhoneInternational!: TuiInputPhoneInternationalPO;

test.beforeEach(({page}) => {
    example = new TuiDocumentationPagePO(page).apiPageExample;
    inputPhoneInternational = new TuiInputPhoneInternationalPO(
        example.locator('tui-textfield'),
    );
});

// [...]

await expect(inputPhoneInternational.dropdown).not.toBeAttached();
await inputPhoneInternational.select.click();
```

File: projects/experimental/components/input-phone-international/input-phone-international.component.ts [hunk 8]
```diff
@@ -0,0 +1,241 @@
...
+    protected readonly search = signal<string>('');
+    protected readonly size = inject(TUI_TEXTFIELD_OPTIONS).size;
+
+    protected readonly mask = tuiMaskito(
+        computed(() => this.computeMask(this.code(), this.metadata())),
+    );
+
+    protected readonly masked = computed(
+        () =>
+            maskitoTransform(this.value(), this.mask() || {mask: /.*/}) || this.el.value,
```
Comment (nsbarsukov) [inline projects/experimental/components/input-phone-international/input-phone-international.component.ts:127]: Should utility `this.computeMask` return always non-nullable value (for code simplicity) ? 🤔 

```ts
import { MASKITO_DEFAULT_OPTIONS } from '@maskito/core';

// [...]

private computeMask(
    countryIsoCode: TuiCountryIsoCode,
    metadata?: MetadataJson,
): MaskitoOptions {
    if (!metadata) {
        return MASKITO_DEFAULT_OPTIONS;
    }

    // [...]
}
```

```ts
protected readonly mask = tuiMaskito(
    computed(() => this.computeMask(this.code(), this.metadata())),
);

protected readonly masked = computed(
    () => maskitoTransform(this.value(), this.mask()) || this.el.value,
);
```

File: projects/experimental/components/input-phone-international/input-phone-international.template.html [hunk 9]
```diff
@@ -0,0 +1,65 @@
...
+            [class.t-ipi-flag_small]="size() === 's'"
+            [src]="code() | tuiFlag"
+        />
+    </button>
+</ng-container>
+<ng-container *tuiTextfieldDropdown>
+    <div
+        *ngIf="countrySearch"
+        #element
+        class="t-ipi-search"
```
Comment (nsbarsukov) [inline projects/experimental/components/input-phone-international/input-phone-international.template.html:28]: Should we hide this gap ?

gap-demo


https://github.com/user-attachments/assets/f80a3c63-300f-4fb8-96bd-cec8af3dacf0

Comment (nsbarsukov) [reply projects/experimental/components/input-phone-international/input-phone-international.template.html:28]: I faced this problem only in Chrome DevTools (Mobile emulation).
**Not** reproducible for IOS safari

Comment (waterplea) [reply projects/experimental/components/input-phone-international/input-phone-international.template.html:28]: Pretty sure it's subpixel gap in non-round scale in emulator.

File: projects/experimental/components/input-phone-international/input-phone-international.component.ts (hunk 8 above)
Comment (waterplea) [reply projects/experimental/components/input-phone-international/input-phone-international.component.ts:127]: I tried that but seems like something with typings of `tuiMaskito` that allows both signals and values still added `null` to types.

Comment (nsbarsukov) [reply projects/experimental/components/input-phone-international/input-phone-international.component.ts:127]: Okay

Let's use `MASKITO_DEFAULT_OPTIONS` here

```suggestion
            maskitoTransform(this.value(), this.mask() || MASKITO_DEFAULT_OPTIONS) || this.el.value,
```

**Review by nsbarsukov — APPROVED** [review]

**Pull request merged**